File: server/src/main/java/com/fluffy/exam/api/ExamLikeController.java
package com.fluffy.exam.api;

import com.fluffy.exam.application.ExamLikeService;
import com.fluffy.global.web.Accessor;
import com.fluffy.global.web.Auth;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class ExamLikeController {

    private final ExamLikeService examLikeService;

    @PostMapping("/api/v1/exams/{examId}/like")
    public ResponseEntity<Void> like(
            @PathVariable Long examId,
            @Auth Accessor accessor
    ) {
        examLikeService.like(examId, accessor);

        return ResponseEntity.ok().build();
    }

    @DeleteMapping("/api/v1/exams/{examId}/like")
    public ResponseEntity<Void> unlike(
            @PathVariable Long examId,
            @Auth Accessor accessor
    ) {
        examLikeService.unlike(examId, accessor);

        return ResponseEntity.ok().build();
    }
}

File: server/src/main/kotlin/com/fluffy/exam/ui/ExamLikeController.kt
package com.fluffy.exam.ui

import com.fluffy.exam.application.ExamLikeService
import com.fluffy.global.web.Accessor
import com.fluffy.global.web.Auth
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class ExamLikeController(
    private val examLikeService: ExamLikeService
) {

    @PostMapping("/api/v1/exams/{examId}/like")
    fun like(
        @PathVariable examId: Long,
        @Auth accessor: Accessor
    ): ResponseEntity<Unit> {
        examLikeService.like(examId, accessor)

        return ResponseEntity.ok().build()
    }

    @DeleteMapping("/api/v1/exams/{examId}/like")
    fun unlike(
        @PathVariable examId: Long,
        @Auth accessor: Accessor
    ): ResponseEntity<Unit> {
        examLikeService.unlike(examId, accessor)

        return ResponseEntity.ok().build()
    }
}